Custom hints delegate

ataulm opened the pull request:
Problem:

We had to supply a custom click usage hint "double tap to play movie" but we don't have any custom actions. Previously, we would be required to create an ActionsAccessibilityDelegate with no actions, just to set the usage hint labels, OR duplicate the logic in our app.

Solution:

Extracted a component to set the usage hints, and replace the logic in ActionsAccessibilityDelegate with this extracted component to avoid duplication.

This means we can create the UsageHintsAccessibilityDelegate if we just want to update the usage hints.

Tests:

No tests since the delegate is called from an overridden android method, it's not testable without robolectric/on a device.

ataulm committed:
Add custom hints accessibility delegate

- this is useful if you just want to update the click/long click usage hints but have no custom actions so no need to use the other one

ataulm committed:
Add usage hints delegate to the actions delegate

- to avoid duplicating logic

xrigau commented on core/src/main/java/com/novoda/accessibility/UsageHintsAccessibilityDelegate.java:
```diff
@@ -0,0 +1,82 @@
+import static android.support.v4.view.accessibility.AccessibilityNodeInfoCompat.ACTION_CLICK;
+import static android.support.v4.view.accessibility.AccessibilityNodeInfoCompat.ACTION_LONG_CLICK;
+
+class UsageHintsAccessibilityDelegate extends AccessibilityDelegateCompat {
```
This class should be public, right? otherwise users can't use it 🚨

xrigau reviewed (changes requested):
Just 1 comment but the rest LGTM 👍

ataulm commented:
DOH

On Thu, 17 Nov 2016, 12:05 Xavi Rigau, notifications@github.com wrote:

> _@xrigau_ requested changes on this pull request.
> 
> ## Just 1 comment but the rest LGTM 👍
> 
> In
> core/src/main/java/com/novoda/accessibility/UsageHintsAccessibilityDelegate.java
> https://github.com/novoda/accessibilitools/pull/16#pullrequestreview-9011561
> :
> 
> > @@ -0,0 +1,82 @@
> > +package com.novoda.accessibility;
> > +
> > +import android.content.res.Resources;
> > +import android.support.annotation.StringRes;
> > +import android.support.v4.view.AccessibilityDelegateCompat;
> > +import android.support.v4.view.accessibility.AccessibilityNodeInfoCompat;
> > +import android.view.View;
> > +
> > +import static android.support.v4.view.accessibility.AccessibilityNodeInfoCompat.ACTION_CLICK;
> > +import static android.support.v4.view.accessibility.AccessibilityNodeInfoCompat.ACTION_LONG_CLICK;
> > +
> > +class UsageHintsAccessibilityDelegate extends AccessibilityDelegateCompat {
> 
> This class should be public, right? otherwise users can't use it 🚨
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/novoda/accessibilitools/pull/16#pullrequestreview-9011561,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/ACjfG_g5aWzAke21uULoT4kRb0PFd2Rhks5q_EL7gaJpZM4K1MT0
> .

## 

Disclaimer: The information in this e-mail and any attachments is the 
property of Novoda Ltd and is confidential and may be legally privileged. 
This e-mail is intended solely for the person or organisation to which it 
is addressed. Any disclosure, copying or other use of the information by 
any person or organisation who is not the intended recipient is strictly 
prohibited and may be unlawful. If you have received this e-mail in error, 
please inform the sender immediately and delete/destroy this e-mail and any 
copies of it. Novoda Ltd has taken reasonable precautions to minimise the 
risk of any software viruses which may damage your systems, but we advise 
that you take the necessary steps to ensure that no virus contamination is 
suffered. Novoda Ltd does not accept any liability for any loss or damage 
caused by the transmission of any virus.

Novoda Ltd, Company No: 347444, Registered in Scotland Registered 
Office: C/O Alexander Sloan, 38 Cadogan Street, Glasgow, G2 7HF, Scotland.  VAT 
Registration Number  GB 984 2525 93

ataulm committed:
Make UsageHintsA11yDelegate public so clients can use the class

ataulm committed:
Remove unhelpful params from javadoc

- there's no description so they're not helpful
- there's no description because they aren't complex params

xrigau commented:
:shipit: o'clock
